Add single weather-representation endpoint

IvannikovG opened the pull request:
Implemented an endpoint for host/ weather-stations/{stationID}

If there is a station it will be returned as per specification for JSON and CSV. If there is no such station 404 is returned.
Includes two tests for JSON and CSV representation

Commit:
Add local file temporarily

Commit:
Merge branch 'feature/weather-stations' of https://github.com/UdSAES/dwd_data_access into feature/weather-stations

Commit:
Add implementation for a single weather station endpoint

Commit:
Add fix to prevent crash, querying non-existent stationId

MoritzStueber commented on handlers/data_access.js:
```diff
@@ -124,10 +125,79 @@ function getWeatherStations (stationCatalog) {
 }
-
-// TODO @Georgii implement
+// GET Single weather station
```
- empty line above comment on following block of code, not below
- also, pattern for these comments is VERB /path/:pathvar

MoritzStueber commented on sample_data/sample_public_key.pem:
```diff
@@ -0,0 +1,9 @@
+-----BEGIN PUBLIC KEY-----
```
I moved this file to `auth/` previously, should only exist there (looks like an accidental commit?) -> delete here, including folder

IvannikovG replied:
yes, correct

MoritzStueber committed:
Move variable declarations to top of function

Done by JavaScript anyway ("hoisting"), considered good practice

MoritzStueber committed:
Call `respondWithNotFound` instead of using duplicated code

MoritzStueber committed:
Fix formatting issues

MoritzStueber commented on handlers/data_access.js:
```diff
@@ -124,10 +125,79 @@ function getWeatherStations (stationCatalog) {
+      })
+    }
+
+    const stations = su.findStationsInVicinityOf(stationCatalog, undefined, undefined, undefined)
+    const urlString = req.originalUrl
+    const stationId = urlString.split('/')[2]
```
all variables should be declared at the top of a function -- javascript does this anyway ("hoisting"), improves readability compared to mix of variable- and function declarations here

MoritzStueber commented on handlers/data_access.js:
```diff
@@ -124,10 +125,79 @@ function getWeatherStations (stationCatalog) {
+
+    const station = getStationById(stations, stationId)[0]
+
+    if (station === undefined) {
+      res.set('Content-Type', 'application/problem+json')
+      res.status(404).send({
+        title: 'Not Found',
+        status: 404,
+        detail: 'The requested resource was not found on this server'
+      })
+      log.info('sent `404 Not Found` as response to ' + req.method + '-request on ' + req.path)
+      return
+    }
```
code duplication :anguished:  (fixed in my latest commits)

MoritzStueber commented on test/test_openapi_validation.js:
```diff
@@ -120,4 +120,24 @@ describe('Verify behaviour of API-instance against OAS and/or expectations', fun
+      assert.include(res.headers['content-type'], 'application/json')
+    })
+  })
+
```
the crash you observed when requesting stationId 1050 (a non-existent one) would have been a good reason to write a test for that -> makes it reproducible, ensures that it is eventually fixed for good

IvannikovG replied:
ok added

MoritzStueber reviewed (commented):
- works fine :+1: 
- duplicated code for sending 404 not such a good idea
- best practice is to declare variables at top of function
- you forgot to run the linter :wink: might enforce that later

I already fixed some of the issues (and marked them as resolved in this conversation)
=> please review all changes, remove the public key & we're good to go!

Commit:
Modify branch according to pull request requirements
1). Remove unintended commited folder with sample_key
2). Add test to check if a non-existent station is queried service does not crash, returning 404

Commit:
Fix linting issues
